labirynt.cpp

```diff
@@ -126,36 +127,96 @@ void generateMap(size_t width, size_t height) {
     }
   }
 }
+void pobranie(int a)
+{
+            time_t koniec;
+            char cos;
+            bool b=false;
+            while(true)
+        {
+            cos=getch();
+            if(!b)
+            {
+                koniec= time(NULL) + 1 ;
+                b=true;
+            }
+            if (time(NULL)>=koniec) break;
+            znak[a]+=cos;
+        }
+}
 
 void calibrate() {
-    cerr<<"Tutaj powinna odbyc sie kalibracja klawiszy!\n";
+string napis[4];
+    napis[0]="lewo";
+    napis[1]="prawo";
+    napis[2]="gore";
+    napis[3]="dol";
+        for(int i=0;i<4;i++)
+        {
+         cout<<"wcisnij strzalke w "<<napis[i]<<" (poczekaj ok. 1s)"<<endl;
+        pobranie(i);
+        }
 }
+void zapisz(int a)
+    {
+        for(int i=0;i<a;i++)
+        {
+        for(int i=9;i<0;i--)
+        {
+            buffor[i-1]=buffor[i];
+        }
+        buffor[0]=getch();
+        }
+    }
 
-char getEvent()
-
+int getEvent()
     {
-        return getch();
-    }// pobierz znak
-void doEvent(char c) {
-    c = toupper(c);
-    unsigned char strzalka =c;//zmienna tylko do strzalek
+        int a=0;
+        string strzalka;
+        int cz=znak[0].length();
+        int b=cz;
+        zapisz(1);
+        switch(buffor[0])
+        {
+        case 'w': a=UP_ARROW_EVENT; break;
+        case 's': a=DOWN_ARROW_EVENT; break;
+        case 'a': a=LEFT_ARROW_EVENT; break;
+        case 'd': a=RIGHT_ARROW_EVENT; break;
+        }
+        if(a==NO_EVENT)
+        {
+            zapisz(1);
+            if (buffor[0]==znak[0][1]) goto dalej;
+            if (buffor[0]==znak[1][1]) goto dalej;
+            if (buffor[0]==znak[2][1]) goto dalej;
+            if (buffor[0]==znak[3][1]) goto dalej;
+            return NO_EVENT;
```

Comment:
Zamiast dawać wszędzie powyżej `goto` mogłeś to zrobić za pomocą `if(){}else if(){}else if(){}[...]else{return NO_EVENT;}`. Zadziałałoby tak samo, a nie musiałbyś etykiet stosować.